File: apps/trace-propagation/java/src/main/java/example/Context.java
package example;

import io.opentracing.util.GlobalTracer;
import io.opentracing.Tracer;
import io.opentracing.SpanContext;

public class Context {

  private static Tracer tracer = GlobalTracer.get();
  public static final String runtime = System.getenv("AWS_EXECUTION_ENV").replace("AWS_Lambda_", "");

  public static String currentTraceId() {
    SpanContext ctx = tracer.activeSpan().context();
    System.out.println("found trace context: traceId=" + ctx.toTraceId() + " spanId=" + ctx.toSpanId());
    return ctx.toTraceId();
  }
}

File: apps/trace-propagation/golang/internal/context.go
package internal

import (
	"context"
	"fmt"
	"os"
	"strconv"
	"strings"

	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
)

var Runtime = strings.ReplaceAll(os.Getenv("AWS_EXECUTION_ENV"), "AWS_Lambda_", "")

type Message struct {
	Runtime string `json:"runtime"`
	TraceID string `json:"trace_id"`
}

func TraceID(ctx context.Context) string {
	span, _ := tracer.SpanFromContext(ctx)
	spanCtx := span.Context()
	spanID, traceID := spanCtx.SpanID(), spanCtx.TraceID()
	fmt.Printf("found span context spanid=%s traceid=%s\n", spanID, traceID)
	return strconv.FormatUint(traceID, 10)
}
